syntax = "proto3";

import "runtime_message.proto";
import "msg_response.proto";

service Runtime{
    rpc Init(InitRequest) returns (MsgResponse) {}
    rpc Done(Empty) returns (MsgResponse) {}
    rpc SetInput (stream SetInputRequest) returns (MsgResponse) {}
    rpc Run (Empty) returns (MsgResponse) {}
    rpc Benchmark (BenchmarkRequest) returns (BenchmarkResponse) {}
    rpc GetOutput (GetOutputRequest) returns (stream GetOutputResponse) {}
    rpc GetInputDetails (Empty) returns (DetailsResponse) {}
    rpc GetOutputDetails (Empty) returns (DetailsResponse) {}
}
